controller/src/main/java/io/pravega/controller/store/stream/InMemoryStream.java

```diff
@@ -959,6 +966,79 @@ private void handleStreamMetadataExists(final long timestamp, CompletableFuture<
         return CompletableFuture.completedFuture(null);
     }
 
+    @Override
+    CompletableFuture<Void> createWriterMarkRecord(String writer, long timestamp, ImmutableMap<Long, Long> position) {
+        CompletableFuture<Void> result = new CompletableFuture<>();
+        WriterMark mark = new WriterMark(timestamp, position);
+
+        synchronized (writersLock) {
+            VersionedMetadata<WriterMark> existing = writerMarks.get(writer);
+            if (existing != null) {
+                result.completeExceptionally(StoreException.create(StoreException.Type.DATA_EXISTS, "writer mark exists"));
```

Comment:
You can just return `new CompletableFuture(...)` or `Futures.failedFuture(...)` instead of creating a `result` variable. That will force whomever will modify this in the future to always return something (there is a possibility that they may just return an incomplete Future).

Reply:
Checkstyle sometimes complains about number of `returns` from a method.

Reply:
There should be a limit, but that limit is not 1. So having one for each branch of the if is ok.